mars-sim-core/src/main/java/com/mars_sim/core/unit/AbstractMobileUnit.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Mars Simulation Project
+ * MobileUnit.java
+ * @date 2024-09-15
+ * @author Barry Evans
+ */
+package com.mars_sim.core.unit;
+
+import com.mars_sim.core.Unit;
+import com.mars_sim.core.UnitType;
+import com.mars_sim.core.location.LocationStateType;
+import com.mars_sim.core.map.location.Coordinates;
+import com.mars_sim.core.map.location.LocalPosition;
+import com.mars_sim.core.structure.Settlement;
+import com.mars_sim.core.structure.building.Building;
+import com.mars_sim.core.vehicle.Vehicle;
+
+/**
+ * Represent a Unit that can move around the surface
+ */
+public abstract class AbstractMobileUnit extends Unit
+	implements MobileUnit  {
+    
+    private Settlement owner;
+    private LocalPosition localPosn = LocalPosition.DEFAULT_POSITION;
+    private int currentBuildingInt;
+
+    /**
+	 * Constructor.
+	 * 
+	 * @param name the name of the unit
+	 * @param owner the unit's location
+	 */
+	protected AbstractMobileUnit(String name, Settlement owner) {
+		super(name, owner.getCoordinates()); 
+        this.owner = owner;
+
+		setContainer(owner);
+	}
+
+	/**
+	 * Set the container of this mobile unit
+	 * @param destination New destination of container
+	 */
+	protected void setContainer(Unit destination) {
+		setContainerID(destination.getIdentifier());
+	}
+
+	/**
+	 * Gets the position at a settlement.
+	 *
+	 * @return distance (meters) from the settlement's center.
+	 */
+	public LocalPosition getPosition() {
+		return localPosn;
+	}
+
+	/**
+	 * Sets the settlement-wide position.
+	 *
+	 * @param position
+	 */
+	public void setPosition(LocalPosition position) {
+		this.localPosn = position;
+	}
+
+    /**
+	 * Computes the building the person is currently located at.
+	 * Returns null if outside of a settlement.
+	 *
+	 * @return building
+	 */
+	@Override
+	public Building getBuildingLocation() {
+		if (currentBuildingInt == -1)
+			return null;
+		return unitManager.getBuildingByID(currentBuildingInt);
+	}
+
+	/**
+	 * Computes the building the person is currently located at.
+	 * Returns null if outside of a settlement.
+	 *
+	 * @return building
+	 */
+	public void setCurrentBuilding(Building building) {
```

Comment:
Never mind. Looks like we still need to call this method. It's simply being relocated to this class.